src/Amazon.Common.DotNetCli.Tools/Utilities.cs

```diff
@@ -288,43 +353,32 @@ public static bool LookPublishAotFlag(string projectLocation, string msBuildPara
                 }
             }
 
-            // If the property wasn't provided in msBuildParameters, fall back to searching project file
-            var projectFile = FindProjectFileInDirectory(projectLocation);
-
-            var xdoc = XDocument.Load(projectFile);
-
-            var element = xdoc.XPathSelectElement("//PropertyGroup/PublishAot");
-            
-            if (bool.TryParse(element?.Value, out bool result))
+            var properties = LookupProjectProperties(projectLocation, "PublishAot");
+            if (properties.TryGetValue("PublishAot", out var publishAot))
             {
-                return result;
+                return bool.TryParse(publishAot, out var result) && result;
             }
-
             return false;
         }
+
+
         public static bool HasExplicitSelfContainedFlag(string projectLocation, string msBuildParameters)
         {
             if (msBuildParameters != null && msBuildParameters.IndexOf("--self-contained", StringComparison.InvariantCultureIgnoreCase) != -1)
             {
                 return true;
             }
 
-            // If the property wasn't provided in msBuildParameters, fall back to searching project file
-            var projectFile = FindProjectFileInDirectory(projectLocation);
-
-            var xdoc = XDocument.Load(projectFile);
-
-            var element = xdoc.XPathSelectElement("//PropertyGroup/SelfContained");
-
-            if (bool.TryParse(element?.Value, out _))
+            var properties = LookupProjectProperties(projectLocation, "SelfContained");
+            if (properties.TryGetValue("SelfContained", out var selfContained))
             {
-                return true;
+                return bool.TryParse(selfContained, out _);
```

Comment:
I am assuming the reason we are allowed to update this without worrying about backward compatibility is because if user's wanted to set the value as `false` previously, then they would have had to not include this property in project file itself. So since we are updating the logic to read the value now, it will not break anything for those customers, since they would not have had the property in the file anyway.

Reply:
I can't think of a reason users would rely on having `<SelfContainted>false</SelfContainted>` and are more likely to cause confusion. I'm also guess most users are relying on the commandline switch `--self-contained` instead of the msbuild property. So I feel safe make this change.